### PR TITLE
fix: price-feeder: filter out exchange rates which aren't in our acceptlist

### PR DESCRIPTION
**Pull request by adamewozniak**
## Description

Filters out exchange rates that aren't in our accept list before they're submitted for voting.

ref: #413 (Addresses 1)

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added appropriate labels to the PR
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: price-feeder/oracle/oracle.go [hunk 1]
```diff
@@ -293,6 +293,12 @@ func (o *Oracle) checkAcceptList(params oracletypes.Params) {
 			o.logger.Warn().Str("denom", symbol).Msg("price missing for required denom")
 		}
 	}
+	for symbol := range o.prices {
```
Comment (adamewozniak) [inline price-feeder/oracle/oracle.go:296]: Not sure if we want this logic to be in `o.SetPrices()`? Seems like that function should be params-agnostic.

File: price-feeder/go.mod [hunk 2]
```diff
@@ -4,17 +4,20 @@ go 1.17
 
 require (
 	github.com/BurntSushi/toml v0.4.1
-	github.com/cosmos/cosmos-sdk v0.44.5
+	github.com/armon/go-metrics v0.3.10
```
Comment (adamewozniak) [inline price-feeder/go.mod:7]: Pinned to the most recent version to include the `DenomList.Contains` method introduced in #418

File: price-feeder/oracle/oracle.go (hunk 1 above)
Comment (alexanderbez) [reply price-feeder/oracle/oracle.go:296]: Yeah I don't think `checkAcceptList` should be mutative. Let's move this into another function or into `SetPrices` directly.

File: price-feeder/oracle/oracle.go [hunk 3]
```diff
@@ -163,14 +163,20 @@ func (o *Oracle) SetPrices() error {
 
 		priceProvider := o.getOrSetProvider(providerName)
 
+		var acceptedPairs []types.CurrencyPair
 		for _, pair := range currencyPairs {
-			if _, ok := requiredRates[pair.Base]; !ok {
-				requiredRates[pair.Base] = struct{}{}
+			if acceptList.Contains(pair.Base) {
+				acceptedPairs = append(acceptedPairs, pair)
+				if _, ok := requiredRates[pair.Base]; !ok {
```
Comment (adamewozniak) [inline price-feeder/oracle/oracle.go:170]: Moved the `requiredRates` logic here so that ticks wouldn't fail with unaccepted currency pairs

**Review by alexanderbez — APPROVED** [review]
Nice 👌

**Pull request merged**